cmake/modules/extensions.cmake

```diff
@@ -3359,7 +3359,12 @@ function(zephyr_get_scoped output scope variable)
     message(FATAL_ERROR "zephyr_get_scoped(): scope ${scope} doesn't exists.")
   endif()
 
-  get_property(value TARGET ${scope}_scope PROPERTY ${variable})
+  if(CMAKE_SCRIPT_MODE_FILE)
```

Comment:
> if this was a collection of separate files it would be easier to cherry-pick and provide alternatives, but that is yet another sizable effort

another one on my wish list, but before moving all those functions to multiple other locations / individual CMake modules, then we must have proper generated docs, else knowing where to find a given function will become even harder.